modin/pandas/test/test_groupby.py

```diff
@@ -246,6 +256,18 @@ def __call__(self, df):
         return df[self.name]
 
 
+# Constructing the Modin Series in BaseOnPython mode causes a warning that we
+# are defaulting to Pandas. The pytest marker at the top of this file that
+# ignores such warnings doesn't apply to code that generates test params, so
+# catch the warning.
+with (
+    warns_that_defaulting_to_pandas()
+    if get_current_execution() == "BaseOnPython"
+    else contextlib.nullcontext()
+):
+    modin_series = pd.Series([1, 5, 7, 8])
+
+
```

Comment:
Everything except this LGTM, however, I'm not going to complain and block the merge if other reviewers would be satisfied with this approach. The problem I see here is that the `modin_series` object is the same all over the parameters.
```python
# Different objects for every element in the 'by' list
@pytest.mark.parametrize("by", [[pd.Series([..]), pd.Series([..])]])
# Same objects for every element in the 'by' list
@pytest.mark.parametrize("by", [[modin_series, modin_series]])
```
Currently, this doesn't cause any problems, but <i>potentially</i> could.

I see several ways to avoid this way of initialization, none of them are perfect though:
1. Default-to-pandas warning is raised in the Series constructor for `BaseOnPython` execution only. Since we already mark all of the tests for `BaseOnPython` to ignore default-to-pandas warnings, why no to do this in the `conftest.py`. However, this still won't solve the problem if some of the non-`BaseOnPython` execution would start raising default to pandas in the frame's construction someday.
https://github.com/modin-project/modin/blob/72ca672f3e79cb4cdd589692db2d0e19d696da5c/modin/conftest.py#L266-L267
```python
    if execution == BASE_EXECUTION_NAME:
        set_base_execution(BASE_EXECUTION_NAME)
        # Change 'filterwarnings' attribute directly in the pytest's config
        config.addinivalue_line("filterwarnings", "default:.*defaulting to pandas.*:UserWarning")
```
2. Replace explicit Series construction with the Series' getters, which would construct a new Series <i>inside</i> the test function, which shouldn't cause any problems.
```python
@pytest.mark.parametrie("by", [lambda: [pd.Series([..]), pd.Series([...])]])
def test_fn(by):
    # Create Series inside the test function affected by the global pytestmark
    by = by()
    ...
```

As I said, I'm not pushing to do these changes in this PR, let's wait for other reviewers' opinions.

Reply:
@dchigarev This seems reasonable, @mvashishtha what do you think?

Reply:
@dchigarev @devin-petersohn  suggestion 2) gives a way to get a list of `by` values for the parameter, but that loses the benefit of using `pytest.mark.parametrize`, since we'd have to loop through the `by` values. We might as well create all the params in the test body anyway. I could instead make the `by` a list of callables each returning the correct object, but that seems too messy. I prefer 1).

I pushed a commit using 1).